application/src/main/java/org/togetherjava/tjbot/commands/moderation/NoteCommand.java

```diff
@@ -0,0 +1,106 @@
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * This command allows users to write notes about others. Notes are persisted and can be retrieved
+ * using {@link AuditCommand}, like other moderative actions.
+ * <p>
+ * The command fails if the user triggering it is lacking permissions to either write a note about
+ * other users or to write a note about the specific given user (for example a moderator attempting
+ * to write a note about an admin).
+ */
+public final class NoteCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(NoteCommand.class);
+    private static final String USER_OPTION = "user";
+    private static final String CONTENT_OPTION = "content";
+    private static final String ACTION_VERB = "write a note about";
+    private final ModerationActionsStore actionsStore;
+    private final Predicate<String> hasRequiredRole;
+
+    /**
+     * Creates a new instance.
+     *
+     * @param actionsStore used to store actions issued by this command
+     * @param config the config to use for this
+     */
+    public NoteCommand(@NotNull ModerationActionsStore actionsStore, @NotNull Config config) {
+        super("note", "Writes a note about the given user", SlashCommandVisibility.GUILD);
+
+        getData()
+            .addOption(OptionType.USER, USER_OPTION, "The user who you want to write a note about",
+                    true)
+            .addOption(OptionType.STRING, CONTENT_OPTION,
+                    "The content of the note you want to write", true);
+
+        hasRequiredRole = Pattern.compile(config.getSoftModerationRolePattern()).asMatchPredicate();
+        this.actionsStore = Objects.requireNonNull(actionsStore);
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        OptionMapping targetOption = event.getOption(USER_OPTION);
+        Member author = event.getMember();
+        Guild guild = event.getGuild();
+        String content = event.getOption(CONTENT_OPTION).getAsString();
+
+        if (!handleChecks(guild.getSelfMember(), author, targetOption.getAsMember(), content,
+                event)) {
+            return;
+        }
+
+        sendNote(targetOption.getAsUser(), author, content, guild, event);
+    }
+
+    @SuppressWarnings("BooleanMethodNameMustStartWithQuestion")
+    private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
```

Comment:
Instead of suppressing the warning here.
Why not rename the method to something like `doesHandleChecks`?

Naming on boolean methods does make it more easy to understand their purpose without documentation, or reading the signature

Reply:
@Heatmanofurioso the problem is that the name is inaccurate then. The `boolean` return type indicates whether the user passed all the checks (or was it the opposite? 😬) and whether action has been taken (such as printing an error message).

So accurate would be sth like `doesPassChecksAndHandleIfNot` but thats kinda awkward.

That said, we kinda established this `handleXXX` thing as a (probably not-so-good) pattern already. `handle` means "check something and if it doesnt hold, also handle the situation already and send an error message to the user". All the moderation commands use this pattern everywhere already.

In hindsight it would probably have been better to use custom exceptions and move the "handle" part to top-level in the `onSlashCommand` with a try-catch. But it is what it is...

Reply:
I agree with that reasoning. Thanks. Really does make sense for it to be a PR and update everything together, so things are consistent.
Resolving this conversation then